src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3821,6 +3820,279 @@ internal new NamedTypeSymbol CreateNativeIntegerTypeSymbol(bool signed)
+                        csharpReturnType.IsIntegralType() &&
+                        TypeSymbol.Equals(csharpLeftType, csharpRightType, TypeCompareKind.ConsiderEverything))
+                    {
+                        return;
+                    }
+                }
+
+                // void* has several comparison operators built in.
+                if (binaryKind is BinaryOperatorKind.Equal or
+                                  BinaryOperatorKind.NotEqual or
+                                  BinaryOperatorKind.GreaterThan or
+                                  BinaryOperatorKind.LessThan or
+                                  BinaryOperatorKind.GreaterThanOrEqual or
+                                  BinaryOperatorKind.LessThanOrEqual &&
+                    csharpReturnType.SpecialType == SpecialType.System_Boolean &&
+                    csharpLeftType is PointerTypeSymbol { PointedAtType.SpecialType: SpecialType.System_Void } &&
+                    csharpRightType is PointerTypeSymbol { PointedAtType.SpecialType: SpecialType.System_Void })
+                {
+                    return;
+                }
+
+                // T* operator+(T*, int i)
+                if (binaryKind is BinaryOperatorKind.Addition &&
+                    csharpLeftType.IsPointerType() &&
+                    csharpRightType.IsIntegralType() &&
+                    TypeSymbol.Equals(csharpLeftType, csharpReturnType, TypeCompareKind.ConsiderEverything))
+                {
+                    return;
+                }
+
+                // T* operator+(int i, T*)
+                if (binaryKind is BinaryOperatorKind.Addition &&
+                    csharpRightType.IsPointerType() &&
+                    csharpLeftType.IsIntegralType() &&
+                    TypeSymbol.Equals(csharpRightType, csharpReturnType, TypeCompareKind.ConsiderEverything))
+                {
+                    return;
+                }
+
+                // T* operator-(T*, int i)
+                if (binaryKind is BinaryOperatorKind.Subtraction &&
+                    csharpLeftType.IsPointerType() &&
+                    csharpRightType.IsIntegralType() &&
+                    TypeSymbol.Equals(csharpLeftType, csharpReturnType, TypeCompareKind.ConsiderEverything))
+                {
+                    return;
+                }
+
+                // long operator-(T*, T*)
+                if (binaryKind is BinaryOperatorKind.Subtraction &&
+                    csharpLeftType.IsPointerType() &&
+                    csharpReturnType.SpecialType is SpecialType.System_Int64 &&
+                    TypeSymbol.Equals(csharpLeftType, csharpRightType, TypeCompareKind.ConsiderEverything))
+                {
+                    return;
+                }
+
+                // ROS<byte> operator+(ROS<byte>, ROS<byte>). Legal because of utf8 strings.
+                if (binaryKind is BinaryOperatorKind.Addition &&
+                    this.IsReadOnlySpanType(csharpReturnType) &&
```

Comment:
>this.IsReadOnlySpanType(csharpReturnType) [](http://example.com/codeflow?start=20&length=41)

The check doesn't look strict enough. There is no check for byte.

Reply:
whoops!